libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Queue/CSFNetwork+Salesforce.m

```diff
@@ -38,20 +40,53 @@ - (void)setDefaultConnectCommunityId:(NSString *)defaultConnectCommunityId {
 
 - (void)setupSalesforceObserver {
     [[NSNotificationCenter defaultCenter] addObserver:self
-                                             selector:@selector(userAccountManagerDidChangeCurrentUser:)
-                                                 name:SFUserAccountManagerDidChangeCurrentUserNotification
+                                             selector:@selector(userAccountManagerDidChangeUserData:)
+                                                 name:SFUserAccountManagerDidChangeUserDataNotification
+                                               object:nil];
+
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(userAccountManagerDidChangeUser:)
+                                                 name:SFUserAccountManagerDidChangeUserNotification
                                                object:nil];
 }
 
 #pragma mark SFAuthenticationManagerDelegate
+- (void)userAccountManagerDidChangeUser:(NSNotification*)notification {
 
-- (void)userAccountManagerDidChangeCurrentUser:(NSNotification*)notification {
     SFUserAccountManager *accountManager = (SFUserAccountManager*)notification.object;
+    SFUserAccountChange change = (SFUserAccountChange)[notification.userInfo[SFUserAccountManagerUserChangeKey] integerValue];
```

Comment:
I'm having a hard time understanding what this logic is meant to accomplish.  So I'm going to walk through a bit, and see if I can unpack some of this.

I know that, prior to this PR, this was implemented:
```obj-c
- (void)userAccountManagerDidChangeCurrentUser:(NSNotification*)notification {
    SFUserAccountManager *accountManager = (SFUserAccountManager*)notification.object;
    if ([accountManager isKindOfClass:[SFUserAccountManager class]]) {
        if ([accountManager.currentUserIdentity isEqual:self.account.accountIdentity] &&
            ![accountManager.currentCommunityId isEqualToString:self.defaultConnectCommunityId])
        {
            self.defaultConnectCommunityId = accountManager.currentCommunityId;
        }
    }
}
```
My guess is that the original implementation was something of a hack—the only good way to listen to user changes was to listen to _currentUser_ changes, because that's all `SFUserAccountManager` posted on.  Since `CSFNetwork` instances are per-user, it then put the additional sanity guard in: "Is this my user?"  And then proceeded accordingly.

In our new paradigm, we're reporting on user changes for _any_ user.  So "changed current user" listening can simply go away in this class now, in favor of user-specific changes.

And I see what @MikeNachbaur is saying now, with regards to the notification "sender" of `SFUserAccountManagerDidChangeUserDataNotification`: this should really be a notification that the specific changing user "owns".  If we amend the notification to make the user account the owner, then it gives consumers the option to only listen to that user (useful in cases like this one), while still having the `nil` fallback if facilities want to listen to changes for any user.